### PR TITLE
fix(sl): warn ops regarding CAA records

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

Prevent [this](https://www.notion.so/opengov/On-Call-Runbook-ab4bffa2472d47ef9b657242632da7f9?pvs=4#c16acb00f4524e84b1ebd6509fdf82db) from occurring by warning the agency and ops beforehand regarding a possible site launch failure route. 

## Solution

Check for existence of CAA records, and if found, check if they 'trust' Amazon, else ask them to add DNS records that specifically allow for this. 

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [X] No - this PR is backwards compatible with ALL of the following feature flags in this [doc](https://www.notion.so/opengov/Existing-feature-flags-518ad2cdc325420893a105e88c432be5)

## Tests 
set export MOCK_AMPLIFY_DOMAIN_ASSOCIATION_CALLS="true" and add this in server.js
```
const formResponses = [
  {
    submissionId: "",
    requesterEmail: "kishore@open.gov.sg",
    repoName: "kishore-test",
    primaryDomain: "isomer.gov.sg",
    redirectionDomain: "www.isomer.gov.sg",
    agencyEmail: "kishore@open.gov.sg",
  },
  {
    submissionId: "",
    requesterEmail: "kishore@open.gov.sg",
    repoName: "kishore-test-dev-gh",
    primaryDomain: "guide.isomer.gov.sg",
    redirectionDomain: "www.guide.isomer.gov.sg",
    agencyEmail: "kishore@open.gov.sg",
  },
  {
    submissionId: "",
    requesterEmail: "kishore@open.gov.sg",
    repoName: "ogp-jiachen",
    primaryDomain: "nushigh.edu.sg",
    redirectionDomain: "www.nushigh.edu.sg",
    agencyEmail: "kishore@open.gov.sg",
  },
  {
    submissionId: "",
    requesterEmail: "kishore@open.gov.sg",
    repoName: "kishore-test-dev-emil",
    primaryDomain: "asean-ifce.com",
    redirectionDomain: "www.asean-ifce.com",
    agencyEmail: "kishore@open.gov.sg",
  },
]

formsgSiteLaunchRouter.handleSiteLaunchResults(formResponses, "test")

```

the email (in the logs) should be out like this: 
Screenshot 2023-12-22 at 9 47 51 AM

### CODE REVIEW

File: src/routes/formsg/formsgSiteLaunch.ts [hunk 1]
```diff
@@ -350,6 +356,35 @@ export class FormsgSiteLaunchRouter {
...
+            )
+          } else if (caaDigResponse.answer) {
+            const caaRecords = caaDigResponse.answer
+
+            /**
+             * NOTE: If there exists more than one CAA Record, we need to
+             * 1. check if they have whitelisted Amazon CAA
+             * 2. if not, send email to inform them to whitelist Amazon CAA
+             */
+            const hasAmazonCAAWhitelisted = caaRecords.some((record) => {
```
Comment (seaerchin) [inline src/routes/formsg/formsgSiteLaunch.ts:372]: the assumption we're making here is that if there's only one CAA record, it's the whitelisted amazon one right?

File: src/routes/formsg/formsgSiteLaunch.ts [hunk 2]
```diff
@@ -350,6 +356,35 @@ export class FormsgSiteLaunchRouter {
...
+              const isAmazonCAA = TRUSTED_AMPLIFY_CAA_RECORDS.some(
+                (trustedCAA) => trustedCAA === record.value
+              )
+              return isAmazonCAA
+            })
+            if (caaRecords.length > 1 && !hasAmazonCAAWhitelisted) {
+              successResult.addCAARecord = true
+            } else {
+              successResult.addCAARecord = false
```
Comment (seaerchin) [inline src/routes/formsg/formsgSiteLaunch.ts:381]: don't we need the `else if asAmazonCAAWhitelisted` check here?

File: src/routes/formsg/formsgSiteLaunch.ts (hunk 1 above)
Comment (kishore03109) [reply src/routes/formsg/formsgSiteLaunch.ts:372]: not really, the condition to check here is: if it has any CAA record, at least one has to be the amazon one
in the case where there is no CAA record, actually all this is ok already

File: src/routes/formsg/formsgSiteLaunch.ts (hunk 2 above)
Comment (kishore03109) [reply src/routes/formsg/formsgSiteLaunch.ts:381]: oh my conidtion is wrong i think that caused the confusion ps

Comment (kishore03109) [reply src/routes/formsg/formsgSiteLaunch.ts:381]: changed, hopefully this clearer

File: src/routes/formsg/formsgSiteLaunch.ts [hunk 3]
```diff
@@ -350,6 +356,35 @@ export class FormsgSiteLaunchRouter {
...
+            const hasAmazonCAAWhitelisted = caaRecords.some((record) => {
+              const isAmazonCAA = TRUSTED_AMPLIFY_CAA_RECORDS.some(
+                (trustedCAA) => trustedCAA === record.value
+              )
+              return isAmazonCAA
+            })
+            if (caaRecords.length > 0 && !hasAmazonCAAWhitelisted) {
+              successResult.addCAARecord = true
+            } else {
+              successResult.addCAARecord = false
```
Comment (seaerchin) [inline src/routes/formsg/formsgSiteLaunch.ts:381]:
```suggestion
              successResult.addCAARecord = caaRecords.length > 0 && !hasAmazonCAAWhitelisted
```

**Review by seaerchin — APPROVED** [review]

**Pull request merged**